Calypso Build: add more dependencies that need to be transpiled

jeherve opened the pull request:
#### Changes proposed in this Pull Request

The Jetpack plugin relies on Calypso Build, and we've recently ran into some issues with some dependencies:

- Query String (https://github.com/sindresorhus/query-string/issues/189)
- Split on First (https://github.com/sindresorhus/split-on-first/issues/6)
- Strict URI Encode (https://github.com/kevva/strict-uri-encode/issues/6)

Those dependencies now use es6, and consequently need to be transpiled before they can be used.

Since those are fairly general modules, I think it would make sense to patch them in Calypso Build right away; other Calypso Build users may rely on the as well, directly or indirectly.

#### Testing instructions

1. Start with [a local installation of Jetpack](https://github.com/Automattic/jetpack/blob/master/docker/README.md)
2. Run `yarn build` to build the dependencies
3. Edit `node_modules/@automattic/calypso-build/webpack/util.js` with the changes from this PR.
4. Run `yarn build-extensions`
5. Visit your site's block editor in IE11; Jetpack blocks should be available.

jeherve committed:
Calypso Build: add more dependencies that need to be transpiled

bisko reviewed (approved):
The proposed changes do fix the issue in Jetpack's context.

The rest of the experience and look of the block in IE11 looks pretty bad, but that should be addressed in a separate PR:

![Screenshot 2020-07-06 at 15 38 23](https://user-images.githubusercontent.com/184938/86593900-c2133c80-bf9e-11ea-98f5-2a1da99f62d5.png)

pento reviewed (approved):
This fix works for me.

That toolbar misalignment is a Gutenberg issue, see WordPress/gutenberg#21389.

jeherve commented:
@Automattic/team-calypso I'd be happy to get your opinion and review on this. Thank you!

scinos commented on packages/calypso-build/webpack/util.js:
```diff
@@ -85,9 +85,12 @@ const nodeModulesToTranspile = [
 	'react-spring/',
 	'regenerate-unicode-properties/',
 	'regexpu-core/',
+	'split-on-first',
+	'strict-uri-encode',
```
These two are missing the trailing `/`
```suggestion
	'split-on-first/',
	'strict-uri-encode/',
```

jeherve replied:
Good point. Done in a27f05454ac12e19edc507bfe010ef06948157be

scinos reviewed (changes requested):
LGTM, assuming the trailing `/` is added for consistency before merging it to master.

jeherve committed:
Switch to new stable version

david-szabo97 reviewed (commented):
I'd like to have this PR in the next version as well:
https://github.com/Automattic/wp-calypso/pull/45433

What do you think about leaving version as-is (6.2.0) then once this and #45433 is merged, then we create a PR to bump the version?

jeherve committed:
Revert "Switch to new stable version"

This reverts commit 0aafe722b0f91538237fba2887b99bd24b2b9d1b.

jeherve committed:
Stick to version 6.2.0 for now

We'll update to 6.3.0 in the near future:
https://github.com/Automattic/wp-calypso/pull/43907#pullrequestreview-483982410

jeherve commented:
That works for me. I've reverted my version changes, so we should be ready to merge with this one, and you can bump the version after your changes. 👍

sirreal commented:
It would be great to make this happen in Jetpack to catch these build issues ahead of time: https://github.com/Automattic/jetpack/issues/13693

I think we package up that linter as a standalone `@automattic/…` package without much difficulty.

simison commented:
@sirreal @david-szabo97 do we need to run that linter for editor kit plugin as well?

david-szabo97 commented:
@simison Yes! We had the same problem there: https://github.com/Automattic/wp-calypso/pull/45433
